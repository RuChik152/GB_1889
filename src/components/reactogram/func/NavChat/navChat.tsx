import React, { FC, useEffect, useState } from 'react';
import style from './navChat.module.scss';
import {NavChatItem} from "./NavChatItem/navChatItem";

interface ChatItem {
    link: string;
    title: string;
}

interface NavChatProps {
    list: ChatItem[];
}

export const NavChat: FC<NavChatProps> = ({ list }) => {

    return  <ul className={style.list} data-testid="chat_test-id">
        {
            list.map((item, index) => <NavChatItem link={item.link} title={item.title} key={index} />)
        }
        </ul>

}
