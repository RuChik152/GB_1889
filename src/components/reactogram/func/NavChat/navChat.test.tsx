import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { NavChat } from './navChat';

const list = [
  {
    link: '#',
    title: 'test_name',
  },
  {
    link: '#',
    title: 'test_name',
  },
];

describe('NavChat', () => {
  it('Render', () => {
    render(<NavChat list={list} />);
  });

  it('Is element', () => {
    render(<NavChat list={list} />);
    const link = screen.getByTestId('chat_test-id');
    expect(link).toBeInTheDocument();
  })

  it('Snapshot', () => {
    const {asFragment} = render(<NavChat list={list} />);
    expect(asFragment()).toMatchSnapshot();
  })
});
