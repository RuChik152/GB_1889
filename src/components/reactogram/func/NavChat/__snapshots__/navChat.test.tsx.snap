// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavChat Snapshot 1`] = `
<DocumentFragment>
  <ul
    class="list"
    data-testid="chat_test-id"
  >
    <li
      class="list__item"
      data-="true"
    >
      <a
        href="#"
      >
        test_name
      </a>
    </li>
    <li
      class="list__item"
      data-="true"
    >
      <a
        href="#"
      >
        test_name
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
